lib/rack/utils.rb

```diff
@@ -289,6 +289,11 @@ def set_cookie_header(key, value)
           else
             raise ArgumentError, "Invalid :same_site value: #{value[:same_site].inspect}"
           end
+        partitioned =
+          if value[:partitioned]
+            raise ArgumentError, 'Partitioned cookies must be set with `secure`' unless value[:secure]
```

Comment:
`samesite=none` also requires secure cookies, but we don't have the same check.  We should be consistent in these cases.  Additionally, we should decide whether it is better to error, or to implicitly mark the cookie secure.

Reply:
Good point. Seeing that browsers have some variability for cookie requirements, perhaps it's better to remove this check altogether and let the user have full control.

At second glance, `set_cookie_header` does read (and work) more like a lower level method that just puts together the string, so I would prefer **not** to set `secure` implicitly either. Maybe, requirement checks could fit in a future, optional, higher level class that calls `set_cookie_header` internally (i.e. `Cookie.new(**please_check_these_args).header`), but that's probably for another time.

Do you want me to remove the checks from the method and from the test?

Reply:
Probably best to remove the checks now.  If we decide to change the behavior both for Partitioned and SameSite=None, we can do that for both at some later point.